### ISSUE TITLE
Joystick analog input is not consistent between joysticks

### ISSUE THREAD

**Issue by Furyhunter** [STJr/SRB2#38]
MB thread: http://mb.srb2.org/showthread.php?t=40203

To quote @RedEnchilada,

>Like I said earlier, it's not that the game itself doesn't support the full 360-degree movement of analog controllers; it's that the gamepad code is really bad at actually feeding the full analog input into the game, and oftentimes ends up just sending it like a digital input instead.

The analog values are probably not normalized.

**Comment by MonsterIestyn** [comment]
My comment from [the GitLab mirror](http://git.magicalgirl.moe/STJr/SRB2/issues/8)
>I realise no one's talked about this issue for a while now, but it might be useful to note that "gamepadjoystickmove" here is only true if Joystick.bGamepadStyle is on. According to comments in src/i_joy.h, Joystick.bGamepadStyle being on apparently means the joystick uses digital axes? If it's off the joystick uses the usual "analog" code instead, which one can see further down in the function:
>
>
>		if (analogjoystickmove && axis != 0)
		{
			// JOYAXISRANGE should be 1023 (divide by 1024)
			cmd->angleturn = (INT16)(cmd->angleturn - ((axis * angleturn[1]) >> 10)); // ANALOG!
		}

>As another note: in SDL builds of SRB2 at least, joyscale/joyscale2 consvar values of 0 make joysticks turn on this digital control mode. Any other values turn it off.
>
>tl;dr: it's probably intentional, and anyway you're probably also looking at the wrong part of the code

Alam's response to my comment:
>The old SDL 1.2 Joystick API have no means of tell us if the input is a dpad or axis. So I just abuse the scale cvar to allow the user to turn it off/on

**Comment by alama** [comment]
Just to note: let use SDL_JoystickGetDeviceType() to notice what type of joystick input we stop abuse the scale cvar....
